src/View/Shortcodes/EmbedShortcodeProvider.php

```diff
@@ -178,8 +179,37 @@ protected static function linkEmbed($arguments, $href, $title)
      */
     protected static function photoEmbed($arguments, $src)
     {
-        $arguments['src'] = $src;
-        unset($arguments['url']);
-        return HTML::createTag('img', $arguments);
+        $data = [
+            'Arguments' => $arguments,
+            'Attributes' => static::buildAttributeListFromArguments($arguments, ['url']),
+            'Src' => $src
+        ];
+
+        return ArrayData::create($data)->renderWith(self::class . '_photo')->forTemplate();
+    }
+
+    /**
+     * Build a list of HTML attributes from embed arguments - used to preserve backward compatibility
+     *
+     * @deprecated 5.0 Use {$Arguments.name} directly in shortcode templates to access argument values
+     * @param array $arguments List of embed arguments
+     * @param array $exclude List of attribute names to exclude from the resulting list
+     * @return ArrayList
+     */
+    private static function buildAttributeListFromArguments(array $arguments, array $exclude = []): ArrayList
+    {
+        $attributes = ArrayList::create();
+        foreach ($arguments as $key => $value) {
+            if (in_array($key, $exclude)) {
+                continue;
+            }
+
+            $attributes->push(ArrayData::create([
+                'Name' => $key,
+                'Value' => Convert::raw2att($value)
```

Comment:
Have you catered for `$value === true`? It's a pity `getAttributesHTML` is a method on `Form` because that would be great for template usage here.

Reply:
How could it be `true`, though? Doesn't `[shortcode someAtt=true]` get parsed as `(string) 'true'`?

Reply:
It’ll just output it as `attribute="true"` rather than `attribute`, but that’s the same as the existing `HTML::createTag()` behaviour: https://github.com/silverstripe/silverstripe-framework/blob/4/src/View/HTML.php#L58-L64